test/network_interface/ApolloNetworkInterface.ts

```diff
@@ -307,3 +317,56 @@ describe('PersistedQueryNetworkInterface', () => {
     });
   });
 });
+
+describe('addPersistedQueries', () => {
+  class GenericNetworkInterface implements NetworkInterface {
+    public query(originalQuery: Request) {
+      return Promise.resolve(originalQuery);
+    }
+  }
+
+  const egql = new ExtractGQL({ inputFilePath: 'nothing' });
+  const queriesDocument = gql`
+    query {
+      author {
+        firstName
+        lastName
+      }
+    }
+  `;
+
+  const queryMap = egql.createMapFromDocument(queriesDocument);
+
+  const request = {
+    query: gql`
+      query {
+        author {
+          firstName
+          lastName
+        }
+      }
+    `,
+  };
+
+  it('should error with an unmapped query', (done) => {
+    const networkInterface = new GenericNetworkInterface();
+    addPersistedQueries(networkInterface, {});
+    networkInterface.query(request).then(() => {
+      done(new Error('Should not resolve'));
+    }).catch((err) => {
+      assert(err);
+      assert.include(err.message, 'Could not find');
+      done();
+    });
+  });
+
+  it('should pass through a query with the persisted query id', () => {
+    const networkInterface = new GenericNetworkInterface();
+    addPersistedQueries(networkInterface, queryMap);
+    const expectedId = queryMap[getQueryDocumentKey(request.query)];
+    return networkInterface.query(request).then((persistedQuery) => {
+      const id = _.get(persistedQuery, 'id');
```

Comment:
I was getting an error when trying to access .id directly since its not part of return type definition.